src/admin.rs

```diff
@@ -180,6 +181,14 @@ pub trait Reboot {
     fn locked() -> bool;
 }
 
+/// Trait indicating that a value can be used as a status
+pub trait StatusBytes: AsRef<[u8]> {
+    // Set the flag indicating that the random generator could properly be created (`false`) or not (`true`)
+    fn set_random_error(&mut self, value: bool);
+    // Get the flag indicating that the random generator could properly be created (`false`) or not (`true`)
+    fn get_random_error(&self) -> bool;
+}
```

Comment:
Maybe it would be easier to only serialize on demand so we don’t need to parse and modify the bytes representation in the runner.

```suggestion
/// Trait indicating that a value can be used as a status
pub trait Status {
    // Returns the flag indicating that the random generator could properly be created (`false`) or not (`true`)
    fn random_error(&mut self, value: bool) -> &mut Option<bool>;
    // Writes the byte representation of this status value to a vec.
    fn serialize_bytes<const N: usize>(&self, v: &mut Vec<u8, N>);
}
```

Reply:
If I were to do this this way I would probably do it with

```rust
pub trait Status {
    type Serialized: AsRef<[u8]>; 
    fn serialize(&self) -> Self::Serialized;
}